test/spec_utils.rb

```diff
@@ -528,10 +528,37 @@ def initialize(*)
     Rack::Utils.status_code(:ok).must_equal 200
   end
 
+  it "return status code and give deprecation warning for obsolete symbols" do
+    replaced_statuses = {
+      payload_too_large: {status_code: 413, standard_symbol: :content_too_large},
+      unprocessable_entity: {status_code: 422, standard_symbol: :unprocessable_content}
+    }
+    dropped_statuses = {bandwidth_limit_exceeded: 509, not_extended: 510}
+    verbose = $VERBOSE
+    warn_arg = nil
+    Rack::Utils.define_singleton_method(:warn) do |*args|
```

Comment:
Should move this inside  the `begin` and update the `ensure` block to remove the singleton method.